rflx/pyrflx/typevalue.py

```diff
@@ -922,57 +931,38 @@ def bytestring(self) -> bytes:
     def fields(self) -> List[str]:
         return [f.name for f in self._type.fields]
 
-    def _update_accessible_fields(self) -> None:
-        nxt = self._next_field(INITIAL.name)
-        fields: List[str] = []
-        while nxt and nxt != FINAL.name:
+    def _is_valid_opaque_field(self, field: str) -> bool:
+
+        assert isinstance(self._fields[field].typeval, CompositeValue)
+        incoming = self._type.incoming(Field(field))
 
+        for l in incoming:
             if (
-                self.__simplified(self._type.field_condition(Field(nxt))) != TRUE
-                or not self._has_first(nxt)
-                or (
-                    not self._has_length(nxt)
-                    if not isinstance(self._fields[nxt].typeval, OpaqueValue)
-                    else not self._is_valid_opaque_field(nxt)
-                )
+                self.__simplified(l.condition) == TRUE
+                and l.length != UNDEFINED
+                and self._fields[l.source.name].set
             ):
-                break
-
-            fields.append(nxt)
-            nxt = self._next_field(nxt)
-        self.accessible_fields = fields
-
-    def _is_valid_opaque_field(self, field: str) -> bool:
-        if self._get_length_unchecked(field) == UNDEFINED:
-            return False
-
-        for edge in self._type.incoming(Field(field)):
-            if self.__simplified(edge.condition) == TRUE:
-                valid_edge = edge
+                valid_edge = l
                 break
         else:
-            return True
+            return False
 
         return all(
-            [
-                (v.name in self._fields and self._fields[v.name].set) or v.name == "Message"
-                for v in valid_edge.length.variables()
-            ]
+            (v.name in self._fields and self._fields[v.name].set) or v.name == "Message"
+            for v in valid_edge.length.variables()
         )
 
     @property
     def valid_fields(self) -> List[str]:
+        # ISSUE: nedbat/coveragepy#515
```

Comment:
Isn't the issue resolved?

Reply:
Fixed by https://github.com/Componolit/RecordFlux/pull/381/commits/dae735322c7b8e264aac926ecbfde30cfe73cf79.